### PR TITLE
Update script to make it use OpsGenie's REST API

### PR DESCRIPTION
**Pull request by cglrkn**
The script is making API calls to the deprecated OpsGenie Web API, we updated the script to make it use new OpsGenie's REST API.

### CODE REVIEW

File: pkg/services/alerting/notifiers/opsgenie.go [hunk 1]
```diff
@@ -119,14 +121,17 @@ func (this *OpsGenieNotifier) closeAlert(evalContext *alerting.EvalContext) erro
...
 
 	bodyJSON := simplejson.New()
-	bodyJSON.Set("apiKey", this.ApiKey)
-	bodyJSON.Set("alias", "alertId-"+strconv.FormatInt(evalContext.Rule.Id, 10))
+	bodyJSON.Set("source", "Grafana")
 	body, _ := bodyJSON.MarshalJSON()
 
 	cmd := &m.SendWebhookSync{
-		Url:        opsgenieCloseAlertURL,
+		Url:        fmt.Sprintf("%s/%s/close?identifierType=alias", opsgenieAlertURL,"alertId-"+strconv.FormatInt(evalContext.Rule.Id, 10)),
```
Comment (bergquist) [inline pkg/services/alerting/notifiers/opsgenie.go]: I think the string can be replaced with `fmt.Sprintf("%s/alertId-%d", opsgenieAlertURL, evalContext.Rule.Id)`

**Review by bergquist — CHANGES_REQUESTED** [review]
Looks good. Just want some minor adjustments for building the url for closing the alert

**Comment by cglrkn** [comment]
I updated the code according to your request. However, we need to put /close?identifierType=alias to make a successful request. Thus, I build the first part as you suggested and added /close?identifierType=alias at the end.

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
@cglrkn Oh thanks for spotting that! :)

Thank you for contributing to Grafana

File: pkg/services/alerting/notifiers/opsgenie.go [hunk 2]
```diff
@@ -119,14 +121,17 @@ func (this *OpsGenieNotifier) closeAlert(evalContext *alerting.EvalContext) erro
...
 
 	bodyJSON := simplejson.New()
-	bodyJSON.Set("apiKey", this.ApiKey)
-	bodyJSON.Set("alias", "alertId-"+strconv.FormatInt(evalContext.Rule.Id, 10))
+	bodyJSON.Set("source", "Grafana")
 	body, _ := bodyJSON.MarshalJSON()
 
 	cmd := &m.SendWebhookSync{
-		Url:        opsgenieCloseAlertURL,
+		Url:        fmt.Sprintf("%s/alertId-%d/close?identifierType=alias", opsgenieAlertURL, evalContext.Rule.Id)
```
Comment (bergquist) [inline pkg/services/alerting/notifiers/opsgenie.go]: The line is missing a comma at the end. otherwise it wont compile

**Review by bergquist — CHANGES_REQUESTED** [review]

**Comment by bergquist** [comment]
@cglrkn Oh thanks for spotting that! :)

Still a minor change needed for this PR to compile. Will merge once its fixed.

**Comment by cglrkn** [comment]
Added the comma, sorry for missing that.

**Comment by bergquist** [comment]
No worries! 

The build is still failing. Seems like some formatting issue. Please run `gofmt -w ./pkg` in the grafana root folder.

**Comment by cglrkn** [comment]
I run the command. Thanks! @bergquist

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
@cglrkn Awesome! thank you for contributing to Grafana! :)

**Pull request merged**